src/components/WatchList/CoinChart.tsx

```diff
@@ -0,0 +1,157 @@
+import { useChainId } from 'src/hooks';
+import { Field } from 'src/state/pswap/actions';
+import { useSwapActionHandlers } from 'src/state/pswap/hooks';
+import { useTokenPriceData } from 'src/state/ptoken/hooks';
+import { formattedNum, toNiceDateYear } from 'src/utils/charts';
+import useUSDCPrice from 'src/utils/useUSDCPrice';
+import { unwrappedToken } from 'src/utils/wrappedCurrency';
+import { DurationBtns, SelectedCoinInfo, TrackIcons } from './styleds';
+
+type Props = {
+  coin: Token;
+};
+
+export const RedirectContext = React.createContext<boolean>(false);
+
+const CoinChart: React.FC<Props> = ({ coin }) => {
+  const chainId = useChainId();
+
+  const weekFrame = TIMEFRAME.find((value) => value.label === '1W');
+
+  const [timeWindow, setTimeWindow] = useState(
+    weekFrame ||
+      ({} as {
+        description: string;
+        label: string;
+        interval: number;
+        momentIdentifier: string;
+      }),
+  );
+
+  const redirect = useContext(RedirectContext);
+
+  const usdcPrice = useUSDCPrice(coin);
+
+  const { onCurrencySelection } = useSwapActionHandlers(chainId);
+  const onCurrencySelect = useCallback(
+    (currency) => {
+      onCurrencySelection(Field.INPUT, currency);
+    },
+    [onCurrencySelection],
+  );
+
+  const priceData =
+    useTokenPriceData(
+      (coin?.address || '').toLowerCase(),
+      timeWindow?.momentIdentifier,
+      timeWindow?.interval,
+      timeWindow?.label,
+    ) || [];
+
+  const token = unwrappedToken(coin, chainId);
+
+  const priceChart = [...priceData];
+  // add current price in chart
+  if (priceChart.length > 0 && usdcPrice) {
+    const timestampnow = Math.floor(Date.now() / 1000);
+
+    priceChart.push({
+      priceUSD: parseFloat(usdcPrice?.toSignificant(4)),
+      timestamp: `${timestampnow}`,
+    });
+  }
+  return (
+    <Box>
+      <SelectedCoinInfo>
+        <CurrencyLogo currency={token} size={48} />
+        <Box>
+          <Text color="text1" fontSize="24px">
+            {token.symbol}
+          </Text>
+          <Text color="green1" fontSize="16px">
+            ${usdcPrice ? usdcPrice?.toSignificant(4, { groupSeparator: ',' }) : '-'}
+          </Text>
+        </Box>
+        <TrackIcons>
+          <Button
+            variant="primary"
+            backgroundColor="primary"
+            color="black"
+            width={'32px'}
+            height={'32px'}
+            padding="0px"
+            href={`${ANALYTICS_PAGE}/#/token/${coin?.address}`}
+            target="_blank"
+            as="a"
+          >
+            <Link size={12} />
+          </Button>
+          {redirect ? (
+            <Button
+              variant="plain"
+              backgroundColor="oceanBlue"
+              color="white"
+              padding="0px 10px"
+              height="32px"
+              href={`/#${BETA_MENU_LINK.swap}?inputCurrency=${coin?.address}`}
+              target=""
+              as="a"
+            >
+              Trade
```

Comment:
1. Trade Button not working on Dashboard Page
2. Make Trade button visible as per user preferance

Reply:
To make the Trade button work on dashboard page, we must import 'RedirectContext' from '@pangolindex/components' and not from the interface.
That's what I did in my PR on interface